gdal/frmts/gtiff/libtiff/tif_dirwrite.c

```diff
@@ -2905,6 +2960,53 @@ _TIFFRewriteField(TIFF* tif, uint16 tag, TIFFDataType in_datatype,
             TIFFSwabLong8( &entry_offset );
     }
 
+/* -------------------------------------------------------------------- */
+/*      When a dummy tag was written due to TIFFDeferStrileArrayWriting() */
+/* -------------------------------------------------------------------- */
+    if( entry_offset == 0 && entry_count == 0 && entry_type == 0 )
+    {
+        if( tag == TIFFTAG_TILEOFFSETS || tag == TIFFTAG_STRIPOFFSETS )
+        {
+            entry_type = (tif->tif_flags&TIFF_BIGTIFF) ? TIFF_LONG8 : TIFF_LONG; 
+        }
+        else
+        {
+            int write_aslong8 = 1;
+            if( count > 1 && tag == TIFFTAG_STRIPBYTECOUNTS )
```

Comment:
Seems like these could be combined?  Maybe the compiler will do this for you :)

Reply:
not really since in the TIFFTAG_STRIPBYTECOUNTS  case we call TIFFStripSize64(), and in the TIFFTAG_TILEBYTECOUNTS case we call TIFFTileSize64()

Reply:
Looked at that twice and didn't see the difference.  Sorry.